src/bin/rdcore/rootmap.rs

```diff
@@ -307,3 +310,34 @@ fn write_boot_uuid_grub2_dropin<P: AsRef<Path>>(uuid: &str, p: P) -> Result<()>
     std::fs::write(p, format!("set BOOT_UUID=\"{uuid}\"\n"))
         .with_context(|| format!("writing {}", p.display()))
 }
+
+fn get_multipath_kargs(blkinfo: &HashMap<String, String>) -> Result<Vec<String>> {
+    let mut kargs = Vec::new();
+
+    let blkname = blkinfo
+        .get("NAME")
+        .and_then(|s| s.rsplit('/').next())
+        .with_context(|| format!("missing NAME in blkinfo: {:?}", blkinfo))?;
+
+    let mpathd_output = runcmd_output!("multipathd", "show", "maps", "format", "%n %w")?;
```

Comment:
Hmm, `%n` prints the name but we want the sysfs name (`%d`).

I think then this function should take as an argument the device path, we canonicalize that (it becomes `/dev/dm-0`), and then we can take its basename to get `dm-0`, which can then be compared to the `%d` entries.